File: sdks/php/examples/approval_flow.php
<?php

/**
 * Example: Create and respond to an approval.
 */

require_once __DIR__ . '/../vendor/autoload.php';

use Atap\Sdk\ATAPClient;
use Atap\Sdk\Model\ApprovalSubject;

// Create an authenticated client
$client = new ATAPClient([
    'base_url' => 'http://localhost:8080',
    'did' => 'did:web:localhost%3A8080:agent:requester',
    'private_key' => '<base64-encoded-ed25519-seed>',
    'client_secret' => 'atap_...',
]);

// 1. Create an approval request
$subject = new ApprovalSubject(
    type: 'data_access',
    label: 'Access user profile data',
    payload: ['scope' => 'read', 'resource' => '/users/123'],
);

$approval = $client->approvals()->create(
    fromDid: 'did:web:localhost%3A8080:agent:requester',
    toDid: 'did:web:localhost%3A8080:human:approver',
    subject: $subject,
);

echo "Created approval: {$approval->id}\n";
echo "  State: {$approval->state}\n";

// 2. List pending approvals (as the approver)
$approvals = $client->approvals()->list();
echo "\nPending approvals: " . count($approvals) . "\n";

// 3. Respond to the approval
$approved = $client->approvals()->respond($approval->id, 'jws-signature-from-approver');
echo "\nApproval responded: {$approved->state}\n";

// 4. Revoke the approval
$revoked = $client->approvals()->revoke($approval->id);
echo "Approval revoked: {$revoked->state}\n";

File: sdks/java/examples/ApprovalFlow.java
import dev.atap.sdk.ATAPClient;
import dev.atap.sdk.model.Approval;
import dev.atap.sdk.model.ApprovalSubject;

import java.util.Collections;
import java.util.List;

/**
 * Example: Create and respond to an approval request.
 */
public class ApprovalFlow {

    public static void main(String[] args) {
        // Create an authenticated client
        try (ATAPClient client = ATAPClient.builder()
                .baseUrl("http://localhost:8080")
                .did("did:web:localhost%3A8080:agent:requester")
                .privateKey("<base64-encoded-private-key>")
                .clientSecret("atap_<your-secret>")
                .build()) {

            // 1. Create an approval request
            ApprovalSubject subject = new ApprovalSubject(
                    "data_access",
                    "Access user profile data",
                    true, // reversible
                    Collections.singletonMap("resource", "user_profiles")
            );

            Approval approval = client.approvals.create(
                    "did:web:localhost%3A8080:agent:requester",
                    "did:web:localhost%3A8080:human:approver",
                    subject
            );

            System.out.println("Created approval: " + approval.getId());
            System.out.println("State: " + approval.getState());

            // 2. List pending approvals
            List<Approval> pending = client.approvals.list();
            System.out.println("Pending approvals: " + pending.size());

            // 3. Respond to an approval (as the approver)
            // Approval responded = client.approvals.respond(approval.getId(), "jws_signature");

            // 4. Revoke an approval
            // Approval revoked = client.approvals.revoke(approval.getId());
        }
    }
}
